fiftyone/core/collections.py

```diff
@@ -440,6 +442,76 @@ def _get_default_frame_fields(
             include_private=include_private, use_db_fields=use_db_fields
         )
 
+    def get_field(self, path):
+        """Returns the field instance of the provided path, or ``None`` if one
+        does not exist.
+
+        Args:
+            path: a field path
+
+        Returns:
+            a :class:`fiftyone.core.fields.Field` instance or ``None``
+        """
+        keys = path.split(".")
+
+        if not keys:
+            return None
+
+        if self.media_type == fom.VIDEO and keys[0] == "frames":
+            schema = self.get_frame_field_schema()
+            keys = keys[1:]
+            field = fofr.Frames
+        else:
+            schema = self.get_field_schema()
+            field = None
+
+        last = len(keys) - 1
+        _add_mapped_fields_as_private_fields(schema)
+        for idx, field_name in enumerate(keys):
+            field = schema.get(field_name, None)
+            if field is None:
+                return field
+
+            if idx != last and isinstance(field, (fof.ListField)):
+                field = field.field
+
+            if isinstance(field, fof.EmbeddedDocumentField):
+                schema = field.get_field_schema()
+                _add_mapped_fields_as_private_fields(schema)
+
+        return field
+
+    def _resolve_field(self, path):
```

Comment:
`SampleCollection._resolve_field()` will translate regular paths to database paths, e.g. `detection.id` to `detection._id`.